libse/SubtitleFormats/Ultech130.cs

```diff
@@ -189,26 +183,36 @@ private static void WriteTime(FileStream fs, TimeCode timeCode)
 
         public override bool IsMine(List<string> lines, string fileName)
         {
-            if (!string.IsNullOrEmpty(fileName) && File.Exists(fileName))
+            // invalid extension
+            if (!(fileName.EndsWith(".ult", StringComparison.Ordinal) || fileName.EndsWith(".uld", StringComparison.Ordinal)))
+            {
+                return false;
+            }
+            // file moved or deleted
+            if (!File.Exists(fileName))
+            {
+                return false;
+            }
+            var fi = new FileInfo(fileName);
+            // too short or too big
+            if (fi.Length < 200 || fi.Length >= 1024000)
             {
-                var fi = new FileInfo(fileName);
-                if (fi.Length >= 200 && fi.Length < 1024000) // not too small or too big
+                return false;
+            }
+            var buffer = new byte[UltechId.Length];
+            using (var fs = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read))
+            {
+                int read = fs.Read(buffer, 0, UltechId.Length);
+                if (read != UltechId.Length)
                 {
-                    if (fileName.EndsWith(".ult", StringComparison.OrdinalIgnoreCase) || fileName.EndsWith(".uld", StringComparison.OrdinalIgnoreCase)) //  drop frame is often named uld, and ult for non-drop
-                    {
-                        byte[] buffer = FileUtil.ReadAllBytesShared(fileName);

```

Comment:
We don't need to read the entire content just to  check ~12 bytes... the file size is also already checked!